### PR TITLE
CONT-568 : Adding deferred function for password

### PR DESCRIPTION
**Pull request by malikparvez**
Adding deferred function for password

### CODE REVIEW

File: manifests/registry.pp [hunk 1]
```diff
@@ -50,6 +56,9 @@
 
   $docker_command = $docker::params::docker_command
 
+  # Use Deferred to encrypt the password
+  $deferred_password = Deferred('docker::deferred_password', [$password])
```
Comment (Ramesh7) [inline manifests/registry.pp:60]: just keep it `docker::password`

File: templates/windows/check_hash.ps1.epp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#file computes the 512SHA for a given string and writes it to a file
```
Comment (jordanbreen28) [inline templates/windows/check_hash.ps1.epp:1]: Shouldnt the corresponding `.erb` files now be removed? like by @Ramesh7 [here](https://github.com/puppetlabs/puppetlabs-sqlserver/pull/436/files)

**Comment by jordanbreen28** [comment]
Looks good.
Could you clean up the commits abit?

File: templates/windows/check_hash.ps1.epp (hunk 2 above)
Comment (malikparvez) [reply templates/windows/check_hash.ps1.epp]: DONE

**Review by GSPatton — APPROVED** [review]
looks good

**Pull request merged**